pkgs/development/python-modules/pykdtree/default.nix

```
@@ -24,6 +24,10 @@ buildPythonPackage rec {
     hash = "sha256-zCCypnxkBWSFoxTSwrbbo1SvfuHI+42uG+byk2o3Q0E=";
   };
 
+  postPatch = ''
+    substituteInPlace pyproject.toml --replace "numpy>=2.0.0rc1,<3" "numpy"
```

Should be a hard failure, IMHO. Otherwise we could miss to remove it when `numpy>=` is available.

```suggestion
    substituteInPlace pyproject.toml \
      --replace-fail "numpy>=2.0.0rc1,<3" "numpy"
```

---

eh, i disagree. if there is a mismatch the build will either succeed because system numpy matches criteria or the build will fail and log an error which will indicate the pattern needs to be updated. adding `--fail` will just cause unnecessary build failures.

---

So `--replace-warn` instead?